src/arch/riscv/inc/arch/platform.h

```diff
@@ -8,6 +8,9 @@
 
 #include <bao.h>
 
+#define PLIC  (1)
+#define APLIC (2)
+
 struct arch_platform {
     paddr_t plic_base;
```

Comment:
The latest versions of the specification have dropped the duo-plic. So finding a platform that implements APLIC and PLIC will be very unlikely. I agree with the first comment, I believe the code becomes more scalable that way. As for the second comment, it seems to me that we can leave the union. I say this because in the version I have with the full AIA I use macros to choose which irqc will be used. Getting something like this in *_desc.c:
```
#if (IRQC == PLIC)
.plic.base = 0xc000000,       
#elif (IRQC == APLIC)
.aia.aplic.base = 0xd000000,
#elif (IRQC == AIA)
.aia.aplic.base = 0xd000000,
.aia.imsic.base = 0x28000000,
#else 
#error "unknown IRQC type " IRQC
#endif
```

What are your thoughts on this?